### DISCUSSION TITLE
Possibility to replace existing recognition model to onnx models exported from other projects?

### DISCUSSION THREAD

**Discussion by lieding** [robertknight/ocrs#68] [General]
Hello, this is a great repository for OCR project. But I find this project now only recognizes English characters, but now I target french text, which is very poor with current recogniton model, maybe. So I consider if it is possible to introduce ONNX model for recognition tasks, detection is the same?

**Comment by robertknight** [comment]
It is indeed possible to swap out the recognition model while keeping the detection model. When using the CLI, the models for different tasks can be specified using the `--detect-model` and `--rec-model` flags.

The basic process would be:

1. Find or train recognition model (see [ocrs-models repo](https://github.com/robertknight/ocrs-models) for details of the current models)
2. Export recognition model to ONNX
3. Convert ONNX model to .rten format using [rten-convert](https://pypi.org/project/rten-convert/)
4. Specify `--rec-model` flag when calling ocrs CLI (or when configuring `OcrEngine` if you're using the library)

In order for the model to work, the "API" of the model (input shape, output shape and pre/post-processing) needs to be the same. This is where you'll run into issues if you try to take an ONNX model from some unrelated project and try to use it today, as those projects probably do different preprocessing.

As long as the API of another project's models is not too different than how the current models work, I think it should be possible to add some flexibility to this project so you could use them though.

**Comment by federicoemartinez** [comment]
I'm working on a wrapper of ocrs to use it from [PHP](https://github.com/federicoemartinez/php-ocr). 

I would like to be able to load other models so other languages are supported. 

Given that I'm a beginner with rust and with OCR models, could you provide some kind of guide of how to do this? How can I know if the API of the model is compatible?
